src/interp/binary-reader-interp.cc

```diff
@@ -1132,7 +1132,10 @@ Result BinaryReaderInterp::OnReturnCallExpr(Index func_index) {
 
   if (func_index >= num_func_imports()) {
     istream_.Emit(Opcode::InterpAdjustFrameForReturnCall, func_index);
-    istream_.Emit(Opcode::Br, GetFuncOffset(func_index));
+    istream_.Emit(Opcode::Br);
+    // We emit this separately to ensure that the index fixup location
+    // is at the right offset in the instruction stream.
+    istream_.Emit(GetFuncOffset(func_index));
```

Comment:
I can't see how this is different to original code.  

Previously we were calling ` Istream::Emit(Opcode::Enum op, u32 val)` which looks like it does the same as `Istream::Emit(Opcode::Enum op)` + `Istream::Emit(u32 val)`..  maybe I am missing something?

Reply:
The issue is that `GetFuncOffset` has a side effect (appends a new index to `func_fixups_`) that looks up `istream_.end()`. In the original code, the appended location points to the br opcode rather than its argument `func_index` that needs fixing up.

Reply:
Oh, nasty.   How about something more like:  `// We emit this separately to ensure that the fixup generated by GetFuncOffset comes after the Br opcode`

Reply:
Sure, that's fine with me! Pushed a commit revising the comment.